Failed to create mob error

geologicexpansion-1.0.1-1.20.1-fabric.jar
fabric-api-0.91.0+1.20.1.jar

```
Failed to create mob
 com.ishland.uwrad.common.CheckedThreadLocalRandom$1: ThreadLocalRandom accessed from a different thread (owner: Server thread, current: Worker-Main-697)
Possible solutions: Find possible causes in the stack trace below and report to the corresponding mod authors 
 
 (You may make this a fatal warning instead of a hard crash with enforce_safe_world_random_access setting in uwrad.properties)
Caused by: java.util.ConcurrentModificationException: ThreadLocalRandom accessed from a different thread (owner: Server thread, current: Worker-Main-697)
        at com.ishland.uwrad.common.CheckedThreadLocalRandom.handleNotOwner(CheckedThreadLocalRandom.java:51) ~[%5Bserver%5D%20unsafe-world-random-access-detector-1.1.0.jar:?]
        at com.ishland.uwrad.common.CheckedThreadLocalRandom.isSafe(CheckedThreadLocalRandom.java:38) ~[%5Bserver%5D%20unsafe-world-random-access-detector-1.1.0.jar:?]
        at com.ishland.uwrad.common.CheckedThreadLocalRandom.method_43156(CheckedThreadLocalRandom.java:82) ~[%5Bserver%5D%20unsafe-world-random-access-detector-1.1.0.jar:?]
        at net.minecraft.class_6566.method_43048(class_6566.java:33) ~[server-intermediary.jar:?]
        at net.minecraft.class_3532.method_32751(net/minecraft/class_3532.java:734) ~[server-intermediary.jar:?]
        at net.minecraft.class_6019.method_35008(class_6019.java:39) ~[server-intermediary.jar:?]
        at com.blackgear.geologicexpansion.common.entity.duck.Duck.<init>(Duck.java:117) ~[geologicexpansion-1.0.1-1.20.1-fabric.jar:?]
        at net.minecraft.class_1299.method_5883(net/minecraft/class_1299.java:544) ~[server-intermediary.jar:?]
        at net.minecraft.class_1948.method_8661(net/minecraft/class_1948.java:518) ~[server-intermediary.jar:?]
        at net.minecraft.class_3754.method_12107(net/minecraft/class_3754.java:453) ~[server-intermediary.jar:?]
        at net.minecraft.class_2806.method_17033(net/minecraft/class_2806.java:140) ~[server-intermediary.jar:?]
        at net.minecraft.class_2806$class_3768.doWork(class_2806.java:309) ~[server-intermediary.jar:?]
        at net.minecraft.class_2806.method_12154(net/minecraft/class_2806.java:252) ~[server-intermediary.jar:?]
        at net.minecraft.class_3898.method_17225(net/minecraft/class_3898.java:684) ~[server-intermediary.jar:?]
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-6.0.8.jar:?]
        at net.minecraft.class_3898.method_17224(net/minecraft/class_3898.java:673) ~[server-intermediary.jar:?]
        at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[?:?]
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
        at net.minecraft.class_3900.method_17634(class_3900.java:62) ~[server-intermediary.jar:?]
        at net.minecraft.class_3846.method_16907(class_3846.java:91) ~[server-intermediary.jar:?]
        at net.minecraft.class_3846.method_16900(class_3846.java:146) ~[server-intermediary.jar:?]
        at net.minecraft.class_3846.run(class_3846.java:102) ~[server-intermediary.jar:?]
        at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395) ~[?:?]
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
```

--- a/common/src/main/java/com/blackgear/geologicexpansion/common/entity/duck/Duck.java
+++ b/common/src/main/java/com/blackgear/geologicexpansion/common/entity/duck/Duck.java
@@ -114,7 +114,7 @@ public class Duck extends Animal implements FluidWalker {
         super(entityType, level);
         this.eggTime = this.random.nextInt(6000) + 6000;
         this.setPathfindingMalus(BlockPathTypes.WATER, 0.0F);
-        this.discardCooldown = NON_FOOD_DISCARD_COOLDOWN.sample(level.random);
+        this.discardCooldown = NON_FOOD_DISCARD_COOLDOWN.sample(this.random);
     }
 
     public static AttributeSupplier.Builder createAttributes() {
@@ -260,7 +260,7 @@ public class Duck extends Animal implements FluidWalker {
                 if (--this.discardCooldown == 0) {
                     this.setItemSlot(EquipmentSlot.MAINHAND, ItemStack.EMPTY);
                     this.level.broadcastEntityEvent(this, DUCK_FISHING_ANIMATION);
-                    this.discardCooldown = NON_FOOD_DISCARD_COOLDOWN.sample(this.level.random);
+                    this.discardCooldown = NON_FOOD_DISCARD_COOLDOWN.sample(this.random);
 
                     // Decreases the hunts remaining per day
                     postFishingProgress();

